service/matching/matchingEngine.go

```diff
@@ -278,6 +278,7 @@ pollLoop:
 			resp := &h.RecordDecisionTaskStartedResponse{
 				PreviousStartedEventId: nextIDResp.EventId,
 				NextEventId:            nextIDResp.EventId,
+				StickyExecutionEnabled: tCtx.queryTaskInfo.queryRequest.StickyExecutionEnabled,
```

Comment:
rather than passing the flag StickyExecutionEnabled from frontend in the request, you can get it here by replacing the call GetWorkflowExecutionNextEventID() to DescWorkflowExecution().  Moreover, you can also get the workflow type from DescWorkflowExecution() and set it here so you don't need the set workflow type in frontend.

Reply:
ok got the idea.